### PR TITLE
Remove components from draft and move UnderlineNav2 to main bundle

### PR DESCRIPTION
**Pull request by broccolinisoup**
Closes https://github.com/github/primer/issues/2071

### <samp>🤖 Generated by Copilot at 51628ad</samp>

This pull request promotes the draft `UnderlineNav` component with improved design and accessibility features to the main bundle, and moves the deprecated version of the component to the `deprecated/` folder. It also removed the draft entry points of `TreeView` , `NavList`, `SegmentedControl`, `SplitPageLayout`

### Changeset 

* Move UnderlineNav to the deprecated bundle

```diff 
- import {UnderlineNav} from '@primer/react'
+ import {UnderlineNav} from '@primer/react/deprecated'
```

* Move UnderlineNav2 (draft) to the main bundle

```diff 
- import {UnderlineNav} from '@primer/react/drafts'
+ import {UnderlineNav} from '@primer/react'
```

* Remove TreeView from drafts

```diff
- import {TreeView} from '@primer/react/drafts'
+ import {TreeView} from '@primer/react'
```

* Remove SegmentedControl from drafts

```diff
- import {SegmentedControl} from '@primer/react/drafts'
+ import {SegmentedControl} from '@primer/react'
```

* Remove NavList from drafts

```diff
- import {NavList} from '@primer/react/drafts'
+ import {NavList} from '@primer/react'
```

* Remove SplitPageLayout from drafts

```diff
- import {SplitPageLayout} from '@primer/react/drafts'
+ import {SplitPageLayout} from '@primer/react'
```

### CODE REVIEW

**Comment by broccolinisoup** [comment]
I am having a trouble to understand why the jest fails https://github.com/primer/react/actions/runs/5064290353/jobs/9277781802 As far as I understand the `console.error` should still be present for the react versions prior to 18 but according to the test results seems like no console.error should be present anymore? Anyone has any insights here? cc @joshblack as he has done some work around this.

**Comment by joshblack** [comment]
@broccolinisoup feel free to remove it here! I think you should be good to go. The `act()` warning is a weird side-effect of mixing and matching React Testing Library and `react-test-renderer`

**Comment by broccolinisoup** [comment]
Regarding the VRT test failures: https://github.com/primer/react/pull/3444#issuecomment-1636299156

File: .changeset/early-timers-speak.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+```
+
+* Remove SplitPageLayout from drafts
+
+```diff
+- import {SplitPageLayout} from '@primer/react/drafts'
++ import {SplitPageLayout} from '@primer/react'
+```
+
+
```
Comment (broccolinisoup) [inline .changeset/early-timers-speak.md:45]: <!-- Changed components: UnderlineNav, TreeView, SegmentedControl, NavList, SplitPageLayout--!>

**Comment by joshblack** [comment]
@broccolinisoup added back in NavList based on the gatsby conversation, hope that was okay! (And feel free to change back if not)

Is everything good to go from your perspective for this PR? 👀

**Comment by broccolinisoup** [comment]
>@broccolinisoup added back in NavList based on the gatsby conversation, hope that was okay! (And feel free to change back if not)

@joshblack It is absolutely great thanks for that commit!

>Is everything good to go from your perspective for this PR? 👀
Yes! I just resolved the merge conflicts on UnderlineNav and can confirm all the changes in UnderlineNav2 folder in `main` made through to the UnderlineNav folder in `next-major` 🥳 

Once I see the checks green, I am going to merge this! Thanks so much for your help on this PR 🙏🏻

**Pull request merged**